### ISSUE TITLE
Interop tests are failing

### ISSUE THREAD

**Issue by nicolasnoble** [grpc/grpc#7639]
Relevant error from https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/7209/console:

```
/go/src/golang.org/x/oauth2/google/default.go:17:2: cannot find package "cloud.google.com/go/compute/metadata" in any of:
    /usr/local/go/src/cloud.google.com/go/compute/metadata (from $GOROOT)
    /go/src/cloud.google.com/go/compute/metadata (from $GOPATH)
```

Did something change in go-land to make tests failing this way ?

**Comment by menghanl** [comment]
`go get golang.org/x/oauth2` doesn't install `golang.org/x/oauth2/google`, which is needed by interop tests. (golang/oauth2#88)

Adding `go get golang.org/x/oauth2/google` to `tools/dockerfile/interoptest/grpc_interop_go/build_interop.sh` should fix this issue.

Another solution would be `go get google.golang.org/grpc` to install the dependencies first, and replace grpc-go src with the one to be tested later.

**Comment by pgrosu** [comment]
I think we're working on the same issue too in #7475.   I thought that Jenkins would kick automatically to test this.  Do you know what's happening?

Thanks,
~p

**Comment by menghanl** [comment]
Not sure what changed recently. This doesn't seem like a new issue, don't know why it only shows recently.

**Comment by pgrosu** [comment]
@menghanl It's the same issue opened multiple times and worked on by multiple teams, which was happily solved in https://github.com/grpc/grpc/pull/7653 :)

**Comment by adelez** [comment]
Looks like the test starts to pass again: https://grpc-testing.appspot.com/view/Pull%20requests/job/gRPC_interop_pull_requests/

**Comment by adelez** [comment]
So does the master: https://grpc-testing.appspot.com/job/gRPC_interop_master/

**Comment by pgrosu** [comment]
Yup, less is more - it's a good sign :)  Now to the other issues, yay!

**Comment by adelez** [comment]
Happened again: FAILED: Failed to build interop docker images.

**Comment by menghanl** [comment]
Trying another fix in #7655

**Comment by pgrosu** [comment]
@adelez Which PR did it fail in?

**Comment by adelez** [comment]
Any updates on this?

**Comment by adelez** [comment]
It's failing again since https://grpc-testing.appspot.com/job/gRPC_interop_master/8968/console:
+ docker rm -f build_grpc_interop_go_27b9f447-05fc-4582-b0cc-90533ffd2200
build_grpc_interop_go_27b9f447-05fc-4582-b0cc-90533ffd2200
+ exit 1

FAILED: build_docker_go [ret=1, pid=23861]

**Comment by menghanl** [comment]
Can you try the fix in #7655?

**Comment by adelez** [comment]
Yes. #7655 works.
